### PR TITLE
Integrate maps

### LINKED ISSUE #5: Map for selecting location
Instead of just typing the location in, display a map to the user (using OpenStreetMap for example), and take the coordinates of their pin and use that to display weather information.

Make it a 2 way thing, and if a user types a location in, it displays a pin on the map as well!

If you have an API key, let @wrussell1999 know so it can be added to the Netlify website hosting it!

### PR DESCRIPTION
**Pull request by Bhavika-Gianey**
fixes: #5 

**Changes made** 
- Displays current weather location data on a map.
 -  the marker can move on map and the coordinates of the pin will display weather information.

### CODE REVIEW

File: src/components/Map/Map.css [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  width: 100%;
+  display: grid;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  place-items: center;
+  margin: 20px 0;
+
+
+
```
Comment (rajat2502) [inline src/components/Map/Map.css]: let's remove this

**Review by rajat2502 — COMMENTED** [review]
@Bhavika-Gianey looks good to me except the empty spaces thing in some files.

**Review by Pratyush1606 — APPROVED** [review]

File: src/components/CurrentCityWeather/CurrentCityWeather.js [hunk 2]
```diff
@@ -40,18 +47,34 @@ const CurrentCityWeather = () => {
...
+  function setCoordinates(place) {
+    var latitude = place.geometry.location.lat();
+    var longitude = place.geometry.location.lng();
+    var coordinates = {lat: latitude, lon: longitude};
+    setCityCoordinates(coordinates);
+  }
+
+  function getCity(address) {
+    var addressComponents = address.split(",");
+    console.log("city ", addressComponents[0]);
```
Comment (NoahCardoza) [inline src/components/CurrentCityWeather/CurrentCityWeather.js]: Got a stray console.log here!

File: src/components/CurrentCityWeather/CurrentCityWeather.js [hunk 3]
```diff
@@ -3,34 +3,41 @@ import "./CurrentCityWeather.css";
...
-          console.log(result);
           if (result["cod"] !== 200) {
             setIsLoaded(true);
             setError(result);
           } else {
             setIsLoaded(true);
             setError();
             setResults(result);
-            console.log(result);
+            console.log(result.name);
```
Comment (NoahCardoza) [inline src/components/CurrentCityWeather/CurrentCityWeather.js]: And another here.

File: src/components/Map/Map.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  city,
+  setCity,
+  cityCoordinates,
+  setCityCoordinates,
+  map,
+  setMap
+}) => {
+
+  useMapEvent("click", (e) => {
+    console.log(e.latlng);
```
Comment (NoahCardoza) [inline src/components/Map/Map.js]: One more here too.

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Great work, just a few `console.log`'s that I found but that's all!

File: src/components/CurrentCityWeather/CurrentCityWeather.js [hunk 5]
```diff
@@ -40,18 +47,34 @@ const CurrentCityWeather = () => {
...
         setError(err);
       });
-  }, [city]);
+  },[cityCoordinates],[]);
+
+
+  function setCoordinates(place) {
+    var latitude = place.geometry.location.lat();
+    var longitude = place.geometry.location.lng();
+    var coordinates = {lat: latitude, lon: longitude};
```
Comment (NoahCardoza) [inline src/components/CurrentCityWeather/CurrentCityWeather.js]: It's best to use `const` variables wherever possible. When you need the variable to be mutable, `let` is usually better than `var`. To learn more check [this article](https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/) out.

File: src/components/CurrentCityWeather/CurrentCityWeather.js [hunk 6]
```diff
@@ -3,34 +3,41 @@ import "./CurrentCityWeather.css";
...
-        "&units=metric" +
-        "&appid=" +
-        process.env.REACT_APP_APIKEY
+      "https://api.openweathermap.org/data/2.5/weather?lat=" +
+          cityCoordinates.lat +
+          "&lon=" +
+          cityCoordinates.lon +
+          "&units=metric" +
+          "&appid=" +
+          process.env.REACT_APP_APIKEY
```
Comment (NoahCardoza) [inline src/components/CurrentCityWeather/CurrentCityWeather.js]: While this code works fine, I saw this done recently and have been using it in my own code to make things a little easier:

```suggestion
      "https://api.openweathermap.org/data/2.5/weather?"
      + `lat=${cityCoordinates.lat}&`
      + `lon=${cityCoordinates.lon}&`
      + "units=metric&"
      + `appid=${process.env.REACT_APP_APIKEY}`
```

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Now that I have more free time (exams are over) I might be going into a little more detail with my reviews. Everythings looks good so far!

**Comment by Bhavika-Gianey** [comment]
@NoahCardoza , I have made the suggested changes and resolved the merge conflicts.
Please review the changes

**Review by NoahCardoza — APPROVED** [review]
LGTM!

**Comment by NoahCardoza** [comment]
Sorry for the premature "LGTM"! I wrote that after reviewing the code but when I checked the preview, it seems like the marker is still being placed based on the top-left corner. 

After looking through my comments I wonder if I never submitted it... I remember writing it up but maybe I never posted it... 

Basically, when I click, the pin-point of the marker isn't where I clicked, instead the top-left of the marker is placed where I click.

Additionally, while the auto center/zoom is cool, it's a little disorienting, not so much when I'm zoomed out, but when I'm zoomed in and it zooms me out it takes more time to zoom back in just to be zoomed back out again.

Maybe set a threshold so if the user is zoomed in more than 9 points, don't zoom them?

**Comment by Bhavika-Gianey** [comment]
> Sorry for the premature "LGTM"! I wrote that after reviewing the code but when I checked the preview, it seems like the marker is still being placed based on the top-left corner.
> 
> After looking through my comments I wonder if I never submitted it... I remember writing it up but maybe I never posted it...
> 
> Basically, when I click, the pin-point of the marker isn't where I clicked, instead the top-left of the marker is placed where I click.
> 
> Additionally, while the auto center/zoom is cool, it's a little disorienting, not so much when I'm zoomed out, but when I'm zoomed in and it zooms me out it takes more time to zoom back in just to be zoomed back out again.
> 
> Maybe set a threshold so if the user is zoomed in more than 9 points, don't zoom them?

Did the changes.
Please review again.

**Comment by NoahCardoza** [comment]
It seems the map is still zooming out when I am up close and click. Also another merge conflict 😭  I'll try to get this merged in first thing tomorrow morning. If you'd like to start working on another PR while you wait for me go ahead!

**Review by Pratyush1606 — APPROVED** [review]

**Comment by Bhavika-Gianey** [comment]
> It seems the map is still zooming out when I am up close and click. Also another merge conflict 😭 I'll try to get this merged in first thing tomorrow morning. If you'd like to start working on another PR while you wait for me go ahead!

I changed the max zoom-in property. The map will now only be zoomed-in upon change of marker position. The merge conflicts are also resolved 🥵.

**Comment by NoahCardoza** [comment]
LGTM!

**Pull request merged**